fetch-engine.sh

```diff
@@ -20,29 +20,39 @@ fi
 if [ "${AUTOMATIC_ENGINE_MANAGEMENT}" = "True" ]; then
 	echo "OpenRA engine version ${ENGINE_VERSION} is required."
 
-	if [ -d "${ENGINE_DIRECTORY}" ]; then
-		if [ "${CURRENT_ENGINE_VERSION}" != "" ]; then
-			echo "Deleting engine version ${CURRENT_ENGINE_VERSION}."
-		else
-			echo "Deleting existing engine (unknown version)."
+	if command -v git >/dev/null 2>&1; then
+		if [ ! -d "${ENGINE_DIRECTORY}/.git" ]; then
+			rm -rf "${ENGINE_DIRECTORY}"
+			git clone "${AUTOMATIC_ENGINE_SOURCE_GIT}" "${ENGINE_DIRECTORY}"
 		fi
 
-		rm -rf "${ENGINE_DIRECTORY}"
-	fi
+		git --git-dir="${ENGINE_DIRECTORY}/.git" --work-tree="${ENGINE_DIRECTORY}" fetch --all
+		git --git-dir="${ENGINE_DIRECTORY}/.git" --work-tree="${ENGINE_DIRECTORY}" checkout "${ENGINE_VERSION}"
```

Comment:
Note: the purpose of the automatic management is to have a foolproof method that can't be put into a state that requires manual intervention to fix.  From my experience dealing with support requests and from what i've seen and heard from other mods, inexperienced users having the wrong engine version is the single biggest cause of problems for mods.

The default behaviour for the SDK is designed to solve that problem, while giving more experienced modders a back door that gives them the power they need at the cost of a bit more work.